### DISCUSSION TITLE
How to visit expressions with = symbol?

### DISCUSSION THREAD

**Discussion by jsouzapp** [detekt/detekt#4261] [Q&A]
Hello.. I'm trying to avoid the use of some values in my project.
But my detekt custom rule is reporting it as an issue, because it detects only the params.rightMargin, not the entire section of the code.
like the above example:

Ok
`params.rightMargin = R.dimen.custom_name_preffix
`

Not Ok
`params.rightMargin = R.dimen.any_name
`

What works:
`params.setRightMargin(R.dimen.custom_name_preffix)
`

Detekt custom rule visitor:
```
  override fun visitDotQualifiedExpression(expression: KtDotQualifiedExpression) {
        super.visitDotQualifiedExpression(expression)
     //here elements are the params that are verified.... rightMargin, topMargin, etc. as a string list
        elements.forEach { element ->
            if (
                expression.text.contains(element) &&
                !expression.text.contains("R.dimen.custom_name_preffix_")
            ) {
                report(expression)
            }
        }
    }
```

How can I make this work for me?

**Comment by cortinico** [comment]
> Hello.. I'm trying to avoid the use of some values in my project.

Can you clarify? Which kind of values?
If you could provide more examples it would be easier to understand

**Reply by jsouzapp** [reply]
Hi :)

If i put on the code:
`params.rightMargin = R.dimen.custom_name_preffix`

This detekt configuration shows it as a violation.

The expected behavior is to now show it as a violation, since it's using my preffix.

but if i put 
`params.setRightMargin(R.dimen.custom_name_preffix)`

Detekt shows it as a violation.
The problem (I think) is the visitor, where if i place a println() to print the expression.text, it prints only the params.rightMargin, ignoring what comes after.

**Reply by cortinico** [reply]
> If i put on the code:
> params.rightMargin = R.dimen.custom_name_preffix
> This detekt configuration shows it as a violation.
> The expected behavior is to now show it as a violation, since it's using my preffix.
> but if i put
> params.setRightMargin(R.dimen.custom_name_preffix)
> Detekt shows it as a violation.

So are you trying to investigate if all the Android resources have your custom prefix?
Like everything should be `R.dimen.custom_name_...` and all the other resources should be not allowed?

**Reply by jsouzapp** [reply]
Yess!

**Reply by jsouzapp** [reply]
The only issue I'm facing is te detection when:
params.rightMargin = R.dimen.custom_name_preffix

that I think is because of the "=" sign

**Reply by cortinico** [reply]
I believe you should implement this as an Android Lint rule that works on the Resource level (on the XML) and not on the Kotlin code. A Detekt Rule will run "too late" as the resource is already created.

**Reply by jsouzapp** [reply]
> I believe you should implement this as an Android Lint rule that works on the Resource level (on the XML) and not on the Kotlin code. A Detekt Rule will run "too late" as the resource is already created.

Sure, I have an android lint for it too..
But I need specially for the cases that are on kotlin code

**Reply by cortinico** [reply]
>  But I need specially for the cases that are on kotlin code

Android Lint supports also those cases. You don't need a detetk rule as it adds unnecessary complexity.

> Sure, I have an android lint for it too..

Also if you already have a rule that prevents resources with this ID, you should not even have those IDs in the code at all
